### PR TITLE
libphonenumber: new recipe

### PR DESCRIPTION
**Pull request by valgur**
Adds libphonenumber: https://github.com/google/libphonenumber

Google's common Java, C++ and JavaScript library for parsing, formatting, and validating international phone numbers.

I have tested the package with all of the options enabled and disabled. Everything worked as expected, except for:
- `use_re2=True` failing with a compilation error,
- `use_icu_regexp=False` failing when linking test_package.

[Packaging status](https://repology.org/project/libphonenumber/versions)

### CODE REVIEW

File: recipes/libphonenumber/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+required_conan_version = ">=1.56.0 <2 || >=2.0.6"
+
+
+class LibphonenumberConan(ConanFile):
+    name = "libphonenumber"
+    description = "Google's common C++ library for parsing, formatting, and validating international phone numbers"
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/google/libphonenumber"
+    topics = ("phone-numbers",)
```
Comment (jwillikers) [inline recipes/libphonenumber/all/conanfile.py:21]: Maybe `phone` should be a separate topic, too?

```suggestion
    topics = ("phone", "phone-numbers",)
```

**Review by jwillikers — APPROVED** [review]

File: recipes/libphonenumber/all/conanfile.py (hunk 1 above)
Comment (valgur) [reply recipes/libphonenumber/all/conanfile.py]: Added.

**Review by jwillikers — APPROVED** [review]

File: recipes/libphonenumber/all/conanfile.py (hunk 1 above)
Comment (jwillikers) [reply recipes/libphonenumber/all/conanfile.py]: Thanks!

**Review by jwillikers — APPROVED** [review]

**Comment by valgur** [comment]
Thanks, @memsharded! I especially appreciate the effort you put into trying to get the MSVC build working, even if it's not really supported.

**Review by jwillikers — APPROVED** [review]

**Comment by memsharded** [comment]
Probably not worth the effort to make it work in Conan 1, I think I'll revert the ``validate()`` part and exclude Conan 1 builds. I tried it locally and it worked with Linux gcc-11, this is why I also tried in CI.

**Review by jwillikers — APPROVED** [review]

**Review by jwillikers — APPROVED** [review]

**Review by memsharded — APPROVED** [review]

**Comment by barbacar** [comment]
hello guys,@valgur @memsharded Is there a chance to add libphonenumber to Conan1 also?

**Comment by memsharded** [comment]
Hi @barbacar 

I am afraid this wouldn't be a priority, at least for the maintainers, all efforts have shifted now to Conan 2 (released 19 months ago).

**Pull request merged**